libcaf_openssl/test/auth_success.cpp

```diff
@@ -0,0 +1,125 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright (C) 2011 - 2017                                                  *
+ * Dominik Charousset <dominik.charousset (at) haw-hamburg.de>                *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include "caf/config.hpp"
+
+#define CAF_SUITE openssl_authentication_success
+#include "caf/test/unit_test.hpp"
+
+#include <vector>
+#include <sstream>
+#include <utility>
+#include <algorithm>
+
+#include <limits.h>
+#include <stdlib.h>
+
+#include "caf/all.hpp"
+#include "caf/io/all.hpp"
+#include "caf/openssl/all.hpp"
+
+using namespace caf;
+
+namespace {
+
+constexpr char local_host[] = "127.0.0.1";
+
+class config : public actor_system_config {
+public:
+  config() {
+    load<io::middleman>();
+    load<openssl::manager>();
+    add_message_type<std::vector<int>>("std::vector<int>");
+    actor_system_config::parse(test::engine::argc(),
+                               test::engine::argv());
+  }
+
+  static std::string data_dir() {
```

Comment:
Navigating the file system makes testing quite brittle. The location of the binary shouldn't matter. Neither the directory from which you are calling `caf-test`. Is there a way around this in OpenSSL? Can we give it a static string instead of files for this test or maybe use temporary files instead?